setup.py

```diff
@@ -570,26 +570,46 @@ def determine_gl_flags():
 
 def determine_sdl2():
     flags = {}
+    # This is redundant; the only time determine_sdl2 is called is preceded by
+    # verification that this condition is already satisfied
     if not c_options['use_sdl2']:
         return flags
 
+    # Check the environment for a Kivy SDL2 path (all platforms) or for Include
+    # and Lib (intended to catch Windows installation, see #4941)
```

Comment:
I agree that the code is not particularity clear in terms of the larger contexts, but the local code should be clear. The type of comments I wanted to avoid is where they repeat info already in the code. 

E.g. `# SDL2 path was not contained in the environment; attempt to infer it` can be seen from the line above it. That line `sdl2_path = environ.get('KIVY_SDL2_PATH', None)` says the exact same things as the text. Similarly, `# Not on OSX, and/or no sdl2_flags defined` can be gottten by looking at the line above `if platform == 'darwin' and sdl2_flags:`. It's similar with the other comments.